lib/instrumentation/@elastic/elasticsearch.js

```diff
@@ -130,13 +147,16 @@ function parsePath(pathString, method) {
  * elasticsearch
  *
  * @param {object} shim The New Relic datastore-shim
+ * @param {string} pkgVersion elasticSearch version, derived from package.json
  * @returns {Function} captureInstanceAttributes method of shim
  */
-function getConnection(shim) {
-  const connectionPool = this.connectionPool.connections[0]
-  const host = connectionPool.url.host.split(':')
-  const port = connectionPool.url.port || host?.[1]
-  return shim.captureInstanceAttributes(host[0], port)
+function getConnection(shim, pkgVersion) {
+  if (semver.gte(pkgVersion, '7.16.0')) {
```

Comment:
is this needed? the instrumentation doesn't get loaded unless it's gte 7.16.0

Reply:
I wasn't sure if I wanted to rip that out: if we expand instrumentation to an earlier version of 7, we'd need to provide a different handling of the connection. If we don't anticipate going back that far, though, I'm happy to take it out.

Reply:
The 7.x we support is over 2 years old. I think we're safe and stopping there.